src/app/my-portfolio/my-portfolio.component.ts

```diff
@@ -28,7 +28,7 @@ export class MyPortfolioComponent implements OnInit {
     getTransactions() {
         SpinnerUtil.showSpinner();
 
-        this.walletService.getUserWallet().subscribe(walletRes => {
+        this.walletService.walletChange$.subscribe(walletRes => {
```

Comment:
The same thing here. Do not subscribe to `walletChange$`, but on `getUserWallet()`.